x-pack/platform/plugins/private/remote_clusters/public/application/sections/components/remote_cluster_config_steps/remote_cluster_setup_trust/remote_cluster_setup_trust.tsx

```diff
@@ -48,32 +40,65 @@ const i18nTexts = {
   ),
 };
 
-const docLinks = {
-  cert: docs.onPremSetupTrustWithCertUrl,
-  apiKey: docs.onPremSetupTrustWithApiKeyUrl,
-  cloud: docs.cloudSetupTrustUrl,
-};
-
 interface Props {
-  onBack: () => void;
-  onSubmit: () => void;
-  isSaving: boolean;
+  next: (model: string) => void;
+  onSecurityChange: (model: string) => void;
+  cancel?: () => void;
+  currentSecurityModel: string;
 }
 
-export const RemoteClusterSetupTrust = ({ onBack, onSubmit, isSaving }: Props) => {
-  const [isModalVisible, setIsModalVisible] = useState<boolean>(false);
-  const { canUseAPIKeyTrustModel, isCloudEnabled } = useContext(AppContext);
+export const RemoteClusterSetupTrust = ({
+  cancel,
+  next,
+  currentSecurityModel,
+  onSecurityChange,
+}: Props) => {
+  const { canUseAPIKeyTrustModel } = useContext(AppContext);
+  const [securityModel, setSecurityModel] = useState<string>(currentSecurityModel);
+
+  const selectModeButton = (securityModelType: string, testSubj: string) => {
+    return securityModel === securityModelType ? (
+      <EuiButton
+        onClick={() => {
+          setSecurityModel(securityModelType);
+          onSecurityChange(securityModelType);
+        }}
+        color="success"
+        iconType="check"
+        iconSide="left"
+        fullWidth
+        data-test-subj={testSubj}
+      >
+        <FormattedMessage
+          id="xpack.remoteClusters.clusterWizard.trustStep.selected"
+          defaultMessage="Selected"
+        />
+      </EuiButton>
+    ) : (
+      <EuiButton
+        onClick={() => {
+          setSecurityModel(securityModelType);
+          onSecurityChange(securityModelType);
+        }}
+        color="text"
+        fullWidth
+        data-test-subj={testSubj}
+      >
+        <FormattedMessage
+          id="xpack.remoteClusters.clusterWizard.trustStep.docs"
+          defaultMessage="Select"
+        />
+      </EuiButton>
+    );
+  };
```

Comment:
How do you feel about avoiding some of the code repetition with something like?

```suggestion
const selectModeButton = (securityModelType: string, testSubj: string) => {
  const isSelected = securityModel === securityModelType;
  const buttonProps = {
    onClick: () => {
      setSecurityModel(securityModelType);
      onSecurityChange(securityModelType);
    },
    color: isSelected ? 'success': 'text',
    fullWidth: true,
    "data-test-subj": testSubj,
    iconType: isSelected ? 'check': undefined,
    iconSide: isSelected ? 'left': undefined,
  };

  return (
    <EuiButton {...buttonProps}>
      <FormattedMessage
        id={
          isSelected
            ? "xpack.remoteClusters.clusterWizard.trustStep.selected"
            : "xpack.remoteClusters.clusterWizard.trustStep.docs"
        }
        defaultMessage={isSelected ? "Selected" : "Select"}
      />
    </EuiButton>
  );
};
```

Reply:
I updated the EUI component and this is not needed anymore